src/pages/playground/Playground.tsx

```diff
@@ -695,16 +710,27 @@ const Playground: React.FC<PlaygroundProps> = ({ workspaceLocation = 'playground
     [props.playgroundSourceChapter, props.playgroundSourceVariant]
   );
 
+  React.useEffect(() => {
+    // TODO: To migrate the state logic away from playgroundSourceChapter
+    //       and playgroundSourceVariant into the language config instead
+    const languageConfigToSet = getLanguageConfig(
+      props.playgroundSourceChapter,
+      props.playgroundSourceVariant
+    );
+    // Hardcoded for Playground only for now, while we await workspace refactoring
+    // to decouple the SicpWorkspace from the Playground.
+    dispatch(playgroundConfigLanguage(languageConfigToSet));
+  }, [dispatch, props.playgroundSourceChapter, props.playgroundSourceVariant]);
+
+  const languageConfig: SALanguage = useTypedSelector(state => state.playground.languageConfig);
+  const shouldShowDataVisualizer = languageConfig.supports.dataVisualizer ?? false;
+  const shouldShowEnvVisualizer = languageConfig.supports.envVisualizer ?? false;
+  const shouldShowSubstVisualizer = languageConfig.supports.substVisualizer ?? false;
```

Comment:
The `??` operators should not longer be necessary, right?

Reply:
It is still necessary. While `supports` is now enforced, the individual keys of `LanguageFeatures` are still optional. While they are defined for Source languages, at the moment, there is still no such "rule-based" logic for Python and Scheme, so they are still undefined there.

After taking another look at it now, perhaps we should make these keys required as well to make the code cleaner, and just explicitly set their values as false for the non-Source languages. I will do that in a subsequent PR.

Thanks for the comments! Sorry I forgot to request your review again 😅.

Reply:
Ah ok, sounds good!